exercises/binary-search/src/test/java/BinarySearchTest.java

```diff
@@ -13,23 +13,23 @@
 
     @Test
     public void findsAValueInAnArrayWithOneElement() {
-        List<Integer> listOfUnitLength = Collections.singletonList(6);
+        List<Character> listOfUnitLength = Collections.singletonList('6');
```

Comment:
Same question as for `custom-set`; is there a pattern for the generic types being used now?

Reply:
Not really. I've tried to use `Integer`, `Character` and `String` and to make sure I've got a couple of tests using each, but I've not got a strict pattern as I didn't think it mattered too much. I could make it a more strict pattern if you think that would be useful?

Reply:
Nah, I think it's good as-is!

Reply:
Will change it to cycle through `Character`, `String`, `Integer` as it's probably best to be consistent :)

Reply:
As I was changing it I remembered why I didn't do that. Some of them can't use `Character` because they use higher numbers, e.g. `11`... So it's only possible to use `Character` for some of the earlier tests, at least for this exercise. I could make the rest cycle through `String` and `Integer`?

Reply:
I didn't see your ealier comment (didn't refresh the page :P ), sorry! Should I just leave it as it is then? For `custom-set` and `binary-search-tree` too?

Reply:
> As I was changing it I remembered why I didn't do that. Some of them can't use `Character` because they use higher numbers, e.g. `11`... So it's only possible to use `Character` for some of the earlier tests, at least for this exercise. I could make the rest cycle through `String` and `Integer`?

This sounds like a good compromise to me!

Reply:
Okay, will do that :) Will do that for the others too, try to cycle through all three but skip `Character` if not possible :)

Reply:
Sadly, as it turns out `String` causes problems as well since it will sort it lexicographically and we're using numbers in all our test cases... So should I instead try to cycle through all three, skip `Chracter` if not possible and also skip `String` if not possible? And make sure that `Character` and `String` are used at least once?

Reply:
That sounds ideal, yep!

Reply:
Great, will do that then :)